packages/javascript-sdk/src/config/index.ts

```diff
@@ -37,6 +37,8 @@ abstract class Config {
    * @param options The options to use as defaults
    */
   public static set(options: ConfigOptions): void {
+    this.setDefaults(options);
```

Comment:
This is really helpful - thank you! I love getting examples for stuff like this, you've made it really understandable.

I've stepped through the examples you've given above and I agree with what you're saying. The only thing I'm not sure of now is where the function setDefaults should live. Having it in the same file would work I think but triggers a linting warning when it's used: `'setDefaults' was used before it was defined.`. Maybe we should move it out to a utility file after all?

Reply:
Cool, glad it was helpful :)

You can fix that lint warning by moving the function definition _above_ the class within which it is used. Using a function before it's defined will usually work because of function hoisting, but it's generally considered a bad practice. So, linters will warn you when you do it.

Reply:
Makes sense, I've reworked this thanks to your advice